### ISSUE TITLE
aws.GetAvailabilityZones fails when the default aws provider is disabled

### ISSUE THREAD

**Issue by itay** [pulumi/pulumi#9055]
## Hello!

- Vote on this issue by adding a 👍 reaction
- To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already)

## Issue details

This seems to be a regression in 3.25.0 (which includes a bug fix to allow disabling of the `kubernetes` default when using `helmv3`). Specifically, when the default `aws` provider is disabled and you use `aws.GetAvailabilityZones`, you get the following error:

```
    	* rpc error: code = Unknown desc = Default provider for 'aws' disabled. 'aws:index/getAvailabilityZones:getAvailabilityZones' must use an explicit provider.
    exit status 1
```

This works fine in 3.24.1

### Steps to reproduce

Here is a Go program that shows this:

```
package main

import (
	"fmt"

	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"

	"github.com/pulumi/pulumi-aws/sdk/v4/go/aws"
)

func main() {
	pulumi.Run(func(ctx *pulumi.Context) error {
		awsProvider, err := aws.NewProvider(ctx, "r53-aws-provider", &aws.ProviderArgs{
			SkipMetadataApiCheck: pulumi.Bool(false),
		})
		if err != nil {
			return err
		}

		state := "available"
		availableAZ, err := aws.GetAvailabilityZones(ctx, &aws.GetAvailabilityZonesArgs{
			State: &state,
		}, pulumi.Provider(awsProvider))
		if err != nil {
			fmt.Println("ERROR")
			return err
		}

		fmt.Println(availableAZ.Names)

		return nil
	})
}
```

with go.mod:
```
module pulumi_test

go 1.16

require (
	github.com/pulumi/pulumi-aws/sdk/v4 v4.37.5 // indirect
	github.com/pulumi/pulumi/sdk/v3 v3.25.0
)
```

and versions:

```
$ pulumi version
v3.25.0
$ pulumi plugin ls
NAME        KIND      VERSION  SIZE    INSTALLED      LAST USED
aws         resource  4.37.5   350 MB  9 minutes ago  now
aws         resource  4.17.0   276 MB  6 months ago   5 days ago
aws         resource  4.7.0    267 MB  6 months ago   5 days ago
aws         resource  4.6.0    261 MB  3 weeks ago    5 days ago
aws         resource  4.5.1    260 MB  9 months ago   5 days ago
aws         resource  4.0.0    255 MB  9 months ago   1 week ago
aws         resource  1.31.0   222 MB  9 months ago   1 week ago
command     resource  0.0.3    23 MB   1 month ago    6 days ago
kubernetes  resource  3.6.3    73 MB   6 months ago   5 days ago
kubernetes  resource  3.3.1    72 MB   6 months ago   1 week ago
kubernetes  resource  3.2.0    72 MB   9 months ago   5 days ago
kubernetes  resource  3.1.1    72 MB   9 months ago   5 days ago
postgresql  resource  3.0.1    39 MB   9 months ago   5 days ago
random      resource  4.3.1    32 MB   2 days ago     2 days ago
random      resource  4.1.1    30 MB   9 months ago   5 days ago
random      resource  1.7.0    38 MB   9 months ago   1 week ago
tls         resource  4.0.0    30 MB   9 months ago   5 days ago
```

Expected: this should work
Actual: it doesn't work, throws the above error

**Comment by iwahbe** [comment]
Yep, this broke from https://github.com/pulumi/pulumi/issues/8853. I misunderstood the path for default providers vs all providers. Thanks for raising the issue and including a great repro. I'll have a fix shortly.

**Comment by AaronFriel** [comment]
@itay as @iwahbe said, we've got a fix under way, we're just adding tests.

In the meantime the workaround is to not use "disable-default-providers", though if that's not an option, we expect to have a point release out today addressing the root cause.

**Comment by itay** [comment]
All good - we'll just hold off on the 3.25.0 upgrade for now :)

**Comment by AaronFriel** [comment]
We're still testing the 3.25.1 release to make sure it makes our quality standards, aka: making sure all integration tests are green. We're seeing a possible flake on Windows tests and bisecting. Keeping this ticket open as #9067 is merged but not yet released.

**Comment by iwahbe** [comment]
This should be fixed in pulumi `2.25.1`.
